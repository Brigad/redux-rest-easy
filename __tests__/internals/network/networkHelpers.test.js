import {
  getNetworkHelpers,
  setNetworkHelpers,
} from '../../../src/internals/network/networkHelpers';

describe('getNetworkHelpers', () => {
  test('without setting them', () => {
    expect(getNetworkHelpers()).toMatchSnapshot();
  });

  test('after setting them', async () => {
    const tokenName = 'customToken';

    setNetworkHelpers({
      getToken: () => tokenName,
    });

    expect(getNetworkHelpers().getToken()).toBe(tokenName);
    expect((await getNetworkHelpers().requestGET()).headers.Authorization).toBe(
      `Bearer ${tokenName}`,
    );
  });


  test('after setting them - promise token', async () => {
    const tokenName = 'customToken';

    setNetworkHelpers({
      getToken: () => Promise.resolve(tokenName),
    });

    expect(await getNetworkHelpers().getToken()).toBe(tokenName);
    expect((await getNetworkHelpers().requestGET()).headers.Authorization).toBe(
      `Bearer ${tokenName}`,
    );
  });

});
