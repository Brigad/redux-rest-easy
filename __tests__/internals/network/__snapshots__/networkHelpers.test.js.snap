// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getNetworkHelpers without setting them 1`] = `
Object {
  "getToken": [Function],
  "handleError": [Function],
  "handleStatusCode": [Function],
  "requestDELETE": [Function],
  "requestGET": [Function],
  "requestPATCH": [Function],
  "requestPOST": [Function],
  "requestPUT": [Function],
}
`;
